### ISSUE TITLE
Javascript disable button not fast enough

### ISSUE THREAD

**Issue by meshuggahtas** [TASVideos/tasvideos#1135]
There are places where the site should disable/enable a button after something happened, but they are not fast enough to actively disallow/allow clicking them:
- Searching at [User List](https://tasvideos.org/Users/List) --> type in 5 characters, select 4 characters, delete it and press Go. You shouldn't be allowed to press Go with fewer than 5 characters, but the button disable was too slow
- Posting a forum post --> Just click it twice. This happens: https://github.com/TASVideos/tasvideos/issues/1134

**Comment by nattthebear** [comment]
> Searching at [User List](https://tasvideos.org/Users/List)

I can easily reproduce this, but I'm having trouble figuring out what the userlist search is supposed to do at all even when it's working.  It doesn't seem to search the userlist?

> Posting a forum post

I can't reproduce this, even when I disable the disable JS and throttle my network to 2G, and try variations of click timing.

**Comment by adelikat** [comment]
You use the search in users/list by typing into the box.  Then an autocomplete box shows up.  Note that it has a minimum of 3 characters before it does that for you
![image](https://user-images.githubusercontent.com/1679846/156276265-ed896036-f12d-49d1-9198-5d03814df551.png)

**Comment by nattthebear** [comment]
I see.  This feels like nonstandard UX:  When a search bar appears at the top of a paged list/table, isn't the usual expectation that the search searches that table?  But this does not; it's just a quick user find, and isn't actually related to the rest of the content of that page at all.

Regarding the original bug, it does seem to be confusable if you add characters and remove characters while it's searching, so maybe we can do something about that.

**Comment by meshuggahtas** [comment]
Tested again:
- Double clicking: Client side javascript performance dependent on whether how much time elapses between pressing the button and disabling it. 

I have like 15 extensions in chrome (started trying out new tools) so it's easier to do it there. Using Firefox with zero extensions, I still see the window being open, but it's much much more smaller.

The issue should be renamed to reflect that it's not too slow, I just experienced it as such, sorry.

**Comment by nattthebear** [comment]
Please stick to things you can actually do in web browsers, not fake non-existent situations you can simulate with 15 extensions.

I am using Firefox.  If I go to `/Forum/Posts/Create/:id`, type some text in, tab over to the Post button, and then hold down my Enter key, my post is made successfully, I am taken to the page with that post on it, and there's exactly one copy of the post (no double-posting).

**Comment by meshuggahtas** [comment]
Let me rephrase this issue again:
- The bug is javascript not disabling buttons fast enough, letting the user unintentionally and accidentally pressing it multiple times IF they 
- The bug is client side performance dependent, so if someone does work on the computer, it's even easier.
- The bug should work on every platform (android ios tv pcs)
- I can 100% reproduce this on default OS Chrome and Firefox with 0 extensions by pressing the tab to the button and holding Enter. If you are able to do this without reproducing the issue, you probably have very high CPU power and the priority is set to your browser, which is definitely not the case for others.

This bug is the root of a problem, and it's affects the bug I mentioned in the opening post, a rather confusing situation requiring users to press back twice.

I'm not going to argue furthermore whether this is a fake or real issue, it's up to you to do whatever you want with this.

**Comment by nattthebear** [comment]
If I can't reproduce it I can't fix it.  I'll try again later today.

**Comment by Masterjun3** [comment]
I tested this and what's important here is that Firefox and Chrome behaviors differ.

Firefox will actively prevent resubmitting a form, even with javascript disabled. So I'm interested to know how @meshuggahtas even managed to reproduce this on Firefox and conclude it's a javascript problem.

Chrome on the other hand has no problem resubmitting if you manage to press the button again. This can be easily tested by enabling network throttling, submitting a post, then waiting the 750ms until the button is enabled again, then repressing the button. However, at 750ms you can't talk of "accidental" reclicking, so everything is fine here.

**Comment by nattthebear** [comment]
Yes, I can see the blank page in Chrome if I slow down the network enough so that the request isn't finished before the 750ms button reactivate, and then I manually activate the Post button again after that 750ms window but before the request finishes.  But that doesn't match what's being claimed here and isn't really much of a problem.  Tabbing over and holding down Enter never produces a multi-submit for me, and that's using the latest Chrome.

I can also reproduce the issue pretty easily on Chrome if I disable all Javascript; again; doesn't line up at all with what's being claimed here.

> Firefox will actively prevent resubmitting a form, even with javascript disabled.

This matches what I'm seeing.  Even with network slowed to molasses, all JS disabled, button permanently enabled, and me clicking like a madman for 20 seconds, there's never any issue on Firefox.

**Comment by adelikat** [comment]
I didn't realize Firefox actually has has dup submit prevention.  Browsers seem to be finally advancing to be at least the bare minimum of what they should be.  Maybe the double click prevention javascript isn't so important.  It was never intended to be full-proof and of course nonscripters can easily bypass it.  It is just supposed to help with the possibility.

The white screen is from a serverside attempt to block dup requests.  It isn't very graceful by design and so it results in a whitescreen.  Perhaps that could be improved but the more important goal was to not have duplicate posts spamming the forums which seemed to be happening quite frequently

**Comment by meshuggahtas** [comment]
I can reproduce and record my screen with keystrokes and taskmgr when possible at a given OS.

I'm asking for a desired scenario to test on it, for example 
- the live site + forum posting + double-clicking
- live site + sending pm + double-clicking

The feature of preventing accidental double posting works, but the result of showing a white screen and the visitors requiring to press back twice because their own mistake what bugs me.

My other suggestion would be, in case it's possible to implement without any hassle and if there isn't any other viable idea: Avoid processing visitor requests for POST submits multiple times within a short time window and instead redirect them after the 1st time. Short time window as in few seconds, it's up to design.
- If visitor was about double posting, continue to show their latest post.
- If visitor was about editing, continue to result

**Comment by nattthebear** [comment]
Issues with the user list should be fixed.  I won't be looking at issues with forum double posting as I cannot reproduce them nor understand how they could even be happening.

**Comment by adelikat** [comment]
I don't see what we can do about it, more than we already have.  If you have too much stuff caked on to your browser, what can we do?  Plus, this should be bonus anyway.  We shouldn't have to prevent double clicking because users should not be double clicking

**Comment by meshuggahtas** [comment]
Personal technical review and opinion on this issue for the last time, with carefully selected words. Feel free to ignore this message if the issue is already closed as in "not going to change", as this comment doesn't shows any new findings just rephrasing.

**The root of the problem and it's evolution from affecting only the website to affecting only the visitor:**
- The website has a server-side problem which cause resources used up for unintended actions unneccessarily
- The website implements client-side javascript to fix this hole, to avoid "double forum posting"
- This implementation makes visitors who make a mistake to be presented with a white screen. This white screen has zero use for the visitor. They will press back to get where they were. They will be on a POST submit page. They need to press back once again. They are finally on the place where they were supposed to be.

**Why is it a problem for visitors:**
Client-side javascript runs on the visitors machine. Visitor can be watching the site from PC, Mobile or TV. Linux, Windows, MacOS, Android, iOS, TV. Old PC, new PC, mobile phone with many other applications, latest mobile phone with 0 apps. Watch the site while encoding videos or buy a gaming PC only for the website. We have no information on Visitors' performance because there is no such forum/communication channel where they are encouraged to tell they have performance issues with the assumption it will be fixed anytime.

Conclusion and suggestions for future cases of these kind of misdesigns:
- Help the visitor understanding they made a mistake and state what was it and give examples how could it happen
- Understand and optimize the resources being used for **both** the site and visitor. Visitor experience relies on both, not just one of them.
- Understand and deal with the **root of the problem**. Simple example used at Quality Assurance introductions (not word by word, as I last heard this in non English language): A bandage won't solve a broken bone, only gives a slight illusion of a problem gone. At one point, all the stacked bandages will just fall off after the broken bone fully shatters while you just look and think "Maybe more bandages could fix it" instead of "Maybe another kind of fix could fix it".
